src/policy/compiler.rs

```diff
@@ -992,9 +992,21 @@ where
                     }
                 })
                 .collect();
+
             if key_vec.len() == subs.len() && subs.len() <= 20 {
                 insert_wrap!(AstElemExt::terminal(Terminal::Multi(k, key_vec)));
             }
+            // Not a threshold, more optimal to convert it to and()s
+            else if k == subs.len() {
+                let mut policy = subs.first().expect("No sub policy in thresh() ?").clone();
+                for sub in &subs[1..] {
+                    policy = Concrete::And(vec![sub.clone(), policy]);
+                }
+
+                ret = best_compilations(policy_cache, &policy, sat_prob, dissat_prob)?;
```

Comment:
It is probably better to compare the `best_compilations` returned here to the ones we computed using the threshold logic, and actually make sure that we're taking the best ones. It seems intuitively true that nested `and`s will beat `thresh` every time but I'm not sure.

Reply:
Agreed that `best_compilations` makes more sense here. But the current code always produces the most optimal compilation. Consider a particular compilation with all AND's replaced as BOOLAND. 
We would have 

`[E] [W] BOOLAND [W] BOOLAND .....`

The corresponding thresh would be:
`[E] [W] ADD [W] ADD [W] ADD.... k EQUAL`. 

Therefore, there exists at least one compilation that is better than the `thresh` one. In any case, this deserves a comment explaining this scenario.

Reply:
I've expanded a bit more on the comment just above (was not sure how to phrase it though):
> // Not a threshold, it's always more optimal to translate it to and()s as we save the
> // resulting threshold check (N EQUAL) in any case.

Reply:
I think this phrasing is fine